fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/RecordValidationTest.java

```diff
@@ -0,0 +1,238 @@
+            commit(context);
+        }
+        // Validate by primary key
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, formatVersion);
+            validateRecord(store, result.get(0), FDBRecordStoreBase.RecordValidationOptions.RECORD_EXISTS);
+            validateRecord(store, result.get(1), null);
+            context.commit();
+        }
+    }
+
+    @Nonnull
+    public static Stream<Arguments> splitNumberAndFormatVersion() {
+        return Stream.of(0, 1, 2)
+                .flatMap(split -> formatVersions()
+                        .map(version -> Arguments.of(split, version)));
+    }
+
+    @ParameterizedTest(name = "testValidateRecordsMissingSplit [splitNumber = {0}, formatVersion = {1}]")
+    @MethodSource("splitNumberAndFormatVersion")
+    public void testValidateRecordsMissingSplit(int splitNumber, int formatVersion) throws Exception {
+        boolean splitLongRecords = true;
+        // unsplit (short) records have split #0; split records splits start at #1
+        // Use this number to decide which record to operate on
+        int recordNumber = (splitNumber == 0) ? 0 : 1;
+
+        final RecordMetaDataHook hook = getRecordMetaDataHook(splitLongRecords);
+        List<FDBStoredRecord<Message>> result = saveRecords(splitLongRecords, formatVersion, hook);
+        // Delete a split
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, formatVersion);
+            // If operating on the short record, #0 is the only split
+            // If operating on the long record, splits can be 1,2,3
+            final Tuple pkAndSplit = result.get(recordNumber).getPrimaryKey().add(splitNumber);
+            byte[] split = store.recordsSubspace().pack(pkAndSplit);
+            store.ensureContextActive().clear(split);
+            commit(context);
+        }
+
+        // Validate by primary key
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, formatVersion);
+            final FDBRecordStoreBase.RecordValidationOptions expected;
+            // When format version is 3 and the record is a short record, deleting the only split will make the record disappear
+            if ((splitNumber == 0) && (formatVersion == 3)) {
+                expected = FDBRecordStoreBase.RecordValidationOptions.RECORD_EXISTS;
+            } else {
+                expected = FDBRecordStoreBase.RecordValidationOptions.VALID_VALUE;
+            }
+            validateRecord(store, result.get(recordNumber), expected);
+            context.commit();
+        }
+    }
+
+    @ParameterizedTest(name = "testValidateRecordsDeserialize [formatVersion = {0}]")
+    @MethodSource("formatVersions")
+    public void testValidateRecordsDeserialize(int formatVersion) throws Exception {
```

Comment:
We aren't currently trying to support it, but would it be worth adding a quick test where you replace the value with a bunch of 0s or something, and then get back that it wasn't deserializable.

Reply:
Done.

Reply:
Done.